src/PAModel/Serializers/MsAppSerializer.cs

```diff
@@ -227,7 +228,11 @@ public static CanvasDocument Load(Stream streamToMsapp, ErrorContainer errors)
                 // This is debug only. The server checksum is out of date with the client checksum
                 // The main checksum validation that matters is the repack after unpack
 #if DEBUG
-                if (app._checksum.ServerStampedChecksum != null && app._checksum.ServerStampedChecksum != currentChecksum.wholeChecksum)
+                // In case the server stamped checksum on the msapp is older than the client, treat it like its missing checksum.
+                var isNullOrOlderChecksum = app._checksum.ServerStampedChecksum == null
```

Comment:
>app._checksum.ServerStampedChecksum == null [](http://example.com/codeflow?start=44&length=43)

nit: may be cleaner w/ short circuiting.

(app._checksum.ServerStampedChecksum == null) |||
(ChecksumMaker.GetChecksumVersion(app._checksum.ServerStampedChecksum) < ChecksumMaker.Version)